compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -398,6 +405,34 @@ private void findTypeParamInTupleForArray(DiagnosticPos pos, BArrayType expType,
         findTypeParam(pos, expType.eType, tupleElementType, env, resolvedTypes, result);
     }
 
+    private void findTypeParamInUnionForArray(DiagnosticPos pos, BArrayType expType, BUnionType actualType,
+                                              SymbolEnv env, HashSet<BType> resolvedTypes, FindTypeParamResult result) {
+        LinkedHashSet<BType> tupleTypes = new LinkedHashSet<>();
+        for (BType type : actualType.getMemberTypes()) {
+            if (type.tag == TypeTags.ARRAY) {
```

Comment:
I don't think this else branch will ever get hit when all members of the union are arrays. The same goes for the map case below.

Reply:
Removed the else parts : 41a8b883de93e15548d1f90cbc4abeaa63428c69